apps/drec-api/src/pods/organization/organization.service.ts

```diff
@@ -212,6 +235,60 @@ export class OrganizationService {
     await this.sendRoleChangeEmail(organization, userToBeChanged, newRole);
   }
 
+  async setBlockchainAddress(
+    id: number,
+    signedMessage: BindBlockchainAccountDTO['signedMessage'],
+  ): Promise<ISuccessResponse> {
+    if (!signedMessage) {
+      throw new BadRequestException('Signed message is empty.');
+    }
+
+    const organization = await this.findOne(id);
+
+    if (organization.blockchainAccountAddress) {
+      throw new ConflictException(
+        'Organization already has a blockchain address',
+      );
+    }
+
+    const address = await recoverTypedSignatureAddress(
+      this.configService.get<string>('REGISTRATION_MESSAGE_TO_SIGN') ||
+        'I register as D-REC user',
+      signedMessage,
+    );
+
+    return this.updateBlockchainAddress(
+      id,
+      utils.getAddress(address),
+      signedMessage,
+    );
+  }
+
+  async updateBlockchainAddress(
+    orgId: number,
+    address: string,
+    signedMessage?: string,
+  ): Promise<ISuccessResponse> {
+    const organization = await this.findOne(orgId);
+
+    const alreadyExistingOrganizationWithAddress = await this.repository.count({
+      blockchainAccountAddress: address,
+    });
+
+    if (alreadyExistingOrganizationWithAddress > 0) {
+      throw new ConflictException(
+        `This blockchain address has already been linked to a different organization.`,
```

Comment:
This should be secured on database level ( unique constraint) as well, is it?

Reply:
It is